pgmax/fg/graph.py

```diff
@@ -861,14 +977,28 @@ def update(msgs: jnp.ndarray, _) -> Tuple[jnp.ndarray, None]:
                 evidence,
                 wiring.var_states_for_edges,
             )
+            ftov_msgs = vtof_msgs
+
             # Compute new factor to variable messages by message passing
-            ftov_msgs = infer.pass_fac_to_var_messages(
-                vtof_msgs,
-                wiring.factor_configs_edge_states,
-                log_potentials,
-                num_val_configs,
-                temperature,
-            )
+            for (
+                factor_subtype,
+                factor_wiring,
+            ) in wiring.wiring_by_factor_subtype.items():
+                if isinstance(factor_wiring, enumeration.EnumerationWiring):
+                    ftov_msgs = infer.pass_enum_fac_to_var_messages(
+                        ftov_msgs,
+                        factor_wiring.factor_configs_edge_states,
+                        log_potentials,
+                        num_val_configs,
+                        temperature,
+                    )
+                elif isinstance(factor_wiring, logical.LogicalWiring):
+                    ftov_msgs = logical.INFERENCE_BY_SUBTYPES[factor_subtype](
+                        ftov_msgs,
+                        factor_wiring.parents_edge_states,
+                        factor_wiring.children_edge_states,
+                        temperature,
+                    )
```

Comment:
No special processing.

We should have a centralized dictionary, mapping factor types to their corresponding factor to variable updates.

For the function arguments, each factor to variable message updates function takes as input `ftov_msgs`, `log_potentials` and `temperature`. It additionally takes some static arguments (which we already have) and some non-static arguments.

For each wiring, we can have another property `nonstatic_arguments`. For an enumeration wiring it would be just `{'factor_configs_edge_states': self.factor_configs_edge_states}`.

Once we define these, at the beginning of the function we can have
```
arguments = {
	factor_type: {
		'log_potentials': log_potentials,
		'temperature': temperature,
		**static_arguments[factor_type], 
		**wiring[factor_type].nonstatic_arguments,
	}
	for factor_type in wiring
}
```
and you do the updates by
```
for factor_type in FAC_TO_VAR_UPDATES:
	ftov_msgs = FAC_TO_VAR_UPDATES[factor_type](**{'ftov_msgs': ftov_msgs, **arguments[factor_type]})
```
where `FAC_TO_VAR_UPDATES` is defined in some centralized place.